addons/stock/tests/test_stock_flow.py

```diff
@@ -1978,6 +1978,26 @@ def test_picking_scheduled_date_readonlyness(self):
 
         self.assertEqual(f.state, 'confirmed')
 
+    def test_stock_move_with_partner_id(self):
+        """ Ensure that the partner_id of the picking entry is
+        transmitted to the SM upon object creation.
+        """
+        partner = self.env['res.partner'].create({'name': 'Hubert Bonisseur de la Bath'})
+        product = self.env['product.product'].create({'name': 'Un petit coup de polish', 'type': 'product'})
+        wh = self.env['stock.warehouse'].search([('company_id', '=', self.env.company.id)], limit=1)
+
+        f = Form(self.env['stock.picking'], view='stock.view_picking_form')
+        f.partner_id = partner
+        f.picking_type_id = wh.out_type_id
+        with f.move_ids_without_package.new() as move:
+            move.product_id = product
+            move.product_uom_qty = 5
+        picking = f.save()
+
+        self.assertTrue(bool(picking.move_lines))
+        for move in picking.move_lines:
+            self.assertEqual(move.partner_id, partner)
```

Comment:
You can replace the three lines above with
```py
self.assertEqual(picking.move_lines.partner_id, partner)
```

Also, can you create another partner `partner02` to test a partner change on the picking ?
```py
picking.partner_id = partner02
self.assertEqual(picking.move_lines.partner_id, partner02)
```